### ISSUE TITLE
Is it possible to initiate 70b GRPO training with zero2?

### ISSUE THREAD

**Issue by KingsleyHsu** [OpenRLHF/OpenRLHF#851]
Is it possible to initiate 70b GRPO training with zero2? I find that using zero3 for 70b training results in low GPU utilization, whereas using zero3 leads to OOM (Out of Memory) errors. The detailed parameters are as follows:

zero3：
`   python3 -m openrlhf.cli.train_ppo_ray \
    --ref_num_nodes 4 \
    --ref_num_gpus_per_node 8 \
    --reward_num_nodes 0 \
    --reward_num_gpus_per_node 0 \
    --actor_num_nodes 4 \
    --actor_num_gpus_per_node 8 \
    --vllm_num_engines 1 \
    --vllm_tensor_parallel_size 8 \
    --save_path "$CKPT_DIR/checkpoint/$RUN_NAME" \
    --save_steps 10 \
    --save_hf_ckpt \
    --ckpt_path "$CKPT_DIR/ckpt/$RUN_NAME" \
    --logging_steps 1 \
    --eval_steps 1 \
    --train_batch_size 128 \
    --rollout_batch_size 256 \
    --micro_train_batch_size 4 \
    --micro_rollout_batch_size 8 \
    --n_samples_per_prompt 4 \
    --max_epochs 1 \
    --num_episodes 5 \
    --prompt_max_len 1024 \
    --generate_max_len 4096 \
    --zero_stage 3 \
    --bf16 \
    --actor_learning_rate 1e-6 \
    --init_kl_coef 0.01 \
    --pretrain $MODEL_PATH \
    --prompt_data $PROMPT_PATH \
    --input_key problem \
    --label_key final_answer \
    --max_samples 100000 \
    --normalize_reward \
    --gradient_checkpointing \
    --remote_rm_url $REWARD_FUNC_PATH \
    --input_template $TEMPLATE_PATH \
    --advantage_estimator group_norm \
    --flash_attn \
    --vllm_gpu_memory_utilization 0.92 \
    --vllm_sync_backend nccl \
    --packing_samples \
    --adam_offload \
    --use_tensorboard "$CKPT_DIR/tensorboard/$RUN_NAME"`

zero2：
`    python3 -m openrlhf.cli.train_ppo_ray \
    --ref_num_nodes 4 \
    --ref_num_gpus_per_node 8 \
    --reward_num_nodes 0 \
    --reward_num_gpus_per_node 0 \
    --actor_num_nodes 4 \
    --actor_num_gpus_per_node 8 \
    --vllm_num_engines 2 \
    --vllm_tensor_parallel_size 4 \
    --save_path "$CKPT_DIR/checkpoint/$RUN_NAME" \
    --save_steps 10 \
    --save_hf_ckpt \
    --ckpt_path "$CKPT_DIR/ckpt/$RUN_NAME" \
    --logging_steps 1 \
    --eval_steps 1 \
    --train_batch_size 128 \
    --rollout_batch_size 256 \
    --micro_train_batch_size 2 \
    --micro_rollout_batch_size 4 \
    --gradient_accumulation_steps 64 \
    --n_samples_per_prompt 4 \
    --max_epochs 1 \
    --num_episodes 5 \
    --prompt_max_len 1024 \
    --generate_max_len 4096 \
    --zero_stage 2 \
    --bf16 \
    --actor_learning_rate 1e-6 \
    --init_kl_coef 0.01 \
    --pretrain $MODEL_PATH \
    --prompt_data $PROMPT_PATH \
    --input_key problem \
    --label_key final_answer \
    --max_samples 100000 \
    --normalize_reward \
    --gradient_checkpointing \
    --remote_rm_url $REWARD_FUNC_PATH \
    --input_template $TEMPLATE_PATH \
    --advantage_estimator group_norm \
    --flash_attn \
    --vllm_gpu_memory_utilization 0.85 \
    --vllm_enable_prefix_caching \
    --vllm_sync_backend nccl \
    --packing_samples \
    --adam_offload \`
